postgres/src/generic_client.rs

```diff
@@ -144,35 +144,35 @@ impl GenericClient for Client {
         I: IntoIterator<Item = P> + Sync + Send,
         I::IntoIter: ExactSizeIterator,
     {
-        self.query_raw(statement, params).await
+        tokio_postgres::Client::query_raw(self, statement, params).await
     }
 
     async fn prepare(&self, query: &str) -> Result<Statement, Error> {
-        self.prepare(query).await
+        tokio_postgres::Client::prepare(self, query).await
     }
 
     async fn prepare_typed(
         &self,
         query: &str,
         parameter_types: &[Type],
     ) -> Result<Statement, Error> {
-        self.prepare_typed(query, parameter_types).await
+        tokio_postgres::Client::prepare_typed(self, query, parameter_types).await
```

Comment:
According to my comment https://github.com/bikeshedder/deadpool/pull/235#issuecomment-1402194780 this and all the other calls to methods native to `deadpool_postgres::ClientWrapper` and `deadpool_postgres::Transaction` are fine as is. Yet I do prefer the explicit call here as well. It just eliminates one possible source for errors.

Reply:
Also, conformity :)

Reply:
> (...) Yet I do prefer the explicit call here as well. It just eliminates one possible source for errors.

100% agree.

Fun fact: Initially I thought about that infinite recursion myself and when clippy warned about another one but not those ones I was pretty sure it was "alright".

I just didn't touch the implementation as it is supposed to be a 1:1 copy from the code in the `tokio_postgres` repository and wanted to keep it as close as possible to the original source code.

Reply:
I just agreed to my own statement. :rofl: 

I meant to quote your (@yotamofek) statement:

> Anyways, I tend to always use semi or fully qualified method calling in cases where the function name is identical, saves the trouble of having to think too much all the method resolution rules.

100% agree.

Reply:
LOL 😂